File: auth-sdk/typescript/src/settings/session.ts
import { AxiosInstance } from "axios";
import { TokenManager } from "../token-manager";

export class SessionSettings {
  constructor(
    private api: AxiosInstance,
    private tokenManager: TokenManager,
  ) {}

  async getCurrent() {
    try {
      const token = await this.tokenManager.getAuthToken();
      const response = await this.api.get(`/session`, {
        headers: { Authorization: `Bearer ${token}` },
      });
      return { success: true, data: response.data.session };
    } catch (error: any) {
      return {
        success: false,
        error: error.response?.data?.message || error.message,
      };
    }
  }

  async list() {
    try {
      const token = await this.tokenManager.getAuthToken();
      const response = await this.api.get(`/session/list`, {
        headers: { Authorization: `Bearer ${token}` },
      });
      return { success: true, data: response.data.sessions };
    } catch (error: any) {
      return {
        success: false,
        error: error.response?.data?.message || error.message,
      };
    }
  }

  async revoke(sessionID: number) {
    try {
      const token = await this.tokenManager.getAuthToken();
      const response = await this.api.post(
        `/session/revoke`,
        { id: sessionID },
        {
          headers: { Authorization: `Bearer ${token}` },
        },
      );
      return { success: true, data: response.data };
    } catch (error: any) {
      return {
        success: false,
        error: error.response?.data?.message || error.message,
      };
    }
  }

  async revokeOther() {
    try {
      const token = await this.tokenManager.getAuthToken();
      const response = await this.api.post(
        `/session/revokeOther`,
        {},
        {
          headers: { Authorization: `Bearer ${token}` },
        },
      );
      return { success: true, data: response.data };
    } catch (error: any) {
      return {
        success: false,
        error: error.response?.data?.message || error.message,
      };
    }
  }
}

File: auth-sdk/flutter/lib/settings/session.dart
import '../account.dart';
import '../config.dart';
import '../token_manager.dart';

class SessionSettings {
  SessionSettings(this.api, this.tokens);
  final AuthApi api;
  final TokenManager tokens;
  Future<AuthResult<JsonMap>> getCurrent() => _mapped('/session', 'session');
  Future<AuthResult<List<dynamic>>> list() =>
      _mapped('/session/list', 'sessions');
  Future<AuthResult<JsonMap>> revoke(int id) =>
      _call('/session/revoke', {'id': id});
  Future<AuthResult<JsonMap>> revokeOther() =>
      _call('/session/revokeOther', {});
  Future<AuthResult<T>> _mapped<T>(String path, String key) async {
    try {
      final data =
          await api.send('GET', path, headers: authHeaders(await tokens.get()));
      return AuthResult.success(data[key] as T);
    } catch (e) {
      return AuthResult.failure('$e');
    }
  }

  Future<AuthResult<JsonMap>> _call(String path, Object body) async {
    try {
      return AuthResult.success(await api.send('POST', path,
          body: body, headers: authHeaders(await tokens.get())));
    } catch (e) {
      return AuthResult.failure('$e');
    }
  }
}
